api_end_end_testing/src/test/java/tds/user/api/tests/UserApiTests.java

```diff
@@ -0,0 +1,359 @@
+     */
+    private UserInfo createUserOneRoleAssoc(String userEmail) {
+
+        List<RoleAssociation> roleAssociations = new ArrayList<RoleAssociation>();
+        roleAssociations.add(new RoleAssociation("Administrator", "CLIENT", "44886"));
+
+        UserInfo userInfo = new UserInfo(userEmail, "amy", "watson", "800-332-4747", roleAssociations);
+
+        given()
+            .contentType(ContentType.JSON)
+            .header(authHeader)
+            .body(userInfo)
+        .when()
+            .post(uriLocation)
+        .then()
+            .statusCode(201)
+            .header("location", endsWith(uriLocation + "/" + userEmail + "/details"));
+
+        // Execute a GET by email to validate that the user was created
+        given()
+            .contentType(ContentType.JSON)
+            .accept(ContentType.JSON)
+            .header(authHeader)
+        .when()
+            .get(uriLocation + "/" + userEmail + "/details")
+        .then()
+            .statusCode(200)
+            .body("firstName", is(userInfo.getFirstName()))
+            .body("lastName", is(userInfo.getLastName()))
+            .body("phoneNumber", is(userInfo.getPhoneNumber()));
+
+        return userInfo;
+    }
+
+    /*
+     *  Test of Delete User, HTTP DELETE of /rest/external/user, 204 success item found and deleted
+     *  Test of Get User, HTTP GET of /rest/external/user/{email}/details, 404 item not found
+     */
+    private void deleteExistingUser(String userEmail) {
+        // Execute a Delete to remove the user that was just created
+        given()
+            .contentType(ContentType.JSON)
+            .header(authHeader)
+            .body(userEmail)
+        .when()
+            .delete(uriLocation)
+        .then()
+            .statusCode(204);
+    }
+
+    /*
+     * Test of Create User, HTTP POST of /rest/external/user, 201 success item created
+     * Test of Delete User, HTTP DELETE of /rest/external/user, 204 success item found and deleted
+     */
+    @Test
+    public void shouldCreateDeleteUserWithOneRole() {
+        String randomUserEmail = createRandomUserEmail();
+
+        // Create a user with one role association
+        UserInfo userInfo = createUserOneRoleAssoc(randomUserEmail);
+
+        // Execute a GET by email to verify that the user information was created
+        given()
+            .contentType(ContentType.JSON)
+            .accept(ContentType.JSON)
+            .header(authHeader)
+        .when()
+            .get(uriLocation + "/" + randomUserEmail + "/details")
+        .then()
+            .statusCode(200)
+            .body("firstName", is(userInfo.getFirstName()))
+            .body("lastName", is(userInfo.getLastName()))
+            .body("phoneNumber", is(userInfo.getPhoneNumber()));
+
+        // Execute a DELETE to delete the user
+        deleteExistingUser(randomUserEmail);
+
+        // Execute a GET by email to validate user is deleted
+        given()
```

Comment:
I think we could include this in the deleteExistingUser() and have it always be there.  After deleting a user the only real way to tell it was successful is to do this GET and check for a 404.  So I say let's always do this in the delete method.